selftest/uprobe/test_program/test_program.go

```diff
@@ -0,0 +1,19 @@
+package main
+
+import (
+	"os"
+)
+
+//go:noinline
+func testFunction() int {
+	if os.Getenv("SELFTEST_UPROBE_DO_BRANCH") == "y" {
```

Comment:
I have this here to demonstrate attaching to offsets not at the top of a symbol, which could make the argument in favor of changing the API to take an offset instead of a symbol name.